apps/cookbook/src/app/page/side-nav/side-nav.component.html

```diff
@@ -1,122 +1,52 @@
-      <li><a routerLink="showcase/list-load-on-demand" [routerLinkActive]="'is-selected'" class="side-nav__item">List - load on demand</a></li>
-      <li><a routerLink="showcase/list-swipe" [routerLinkActive]="'is-selected'" class="side-nav__item">List - swipe</a></li>
-      <li><a routerLink="showcase/list-no-shape" [routerLinkActive]="'is-selected'" class="side-nav__item">List - No styling</a></li>
-      <li><a routerLink="showcase/reorder-list" [routerLinkActive]="'is-selected'" class="side-nav__item">List - reorder</a></li>
-      <li><a routerLink="showcase/page" [routerLinkActive]="'is-selected'" class="side-nav__item">Page</a></li>
-      <li><a routerLink="showcase/slides" [routerLinkActive]="'is-selected'" class="side-nav__item">Slides</a></li>
-      <li><a routerLink="showcase/tabs" [routerLinkActive]="'is-selected'" class="side-nav__item">Tabs</a></li>
-      <li><a routerLink="showcase/toolbar" [routerLinkActive]="'is-selected'" class="side-nav__item">Toolbar</a></li>
-    </ul>
-  </li>
-  <li routerLinkActive="is-selected" [routerLinkActiveOptions]="{ exact: false }">
-    <a class="side-nav__item side-nav-header">
-      <svg class="icon">
-        <use xlink:href="../../assets/icons/icon-sprite.svg#icon-sleepy"></use>
-      </svg>
-      Other
-    </a>
-    <!-- prettier-ignore -->
-    <ul class="side-nav side-nav--sub">
-      <li><a routerLink="showcase/avatar" [routerLinkActive]="'is-selected'" class="side-nav__item">Avatar</a></li>
-      <li><a routerLink="showcase/badge" [routerLinkActive]="'is-selected'" class="side-nav__item">Badge</a></li>
-      <li><a routerLink="showcase/chart" [routerLinkActive]="'is-selected'" class="side-nav__item">Charts</a></li>
-      <li><a routerLink="showcase/stock-chart" [routerLinkActive]="'is-selected'" class="side-nav__item">Stock Chart</a></li>
-      <li><a routerLink="showcase/colors" [routerLinkActive]="'is-selected'" class="side-nav__item">Colors</a></li>
-      <li><a routerLink="showcase/flag" [routerLinkActive]="'is-selected'" class="side-nav__item">Flag</a></li>
-      <li><a routerLink="showcase/icon" [routerLinkActive]="'is-selected'" class="side-nav__item">Icon</a></li>
-      <li><a routerLink="showcase/progress-circle" [routerLinkActive]="'is-selected'" class="side-nav__item">Progress Circle</a></li>
-      <li><a routerLink="showcase/spinner" [routerLinkActive]="'is-selected'" class="side-nav__item">Spinner</a></li>
-    </ul>
-  </li>
-  <li routerLinkActive="is-selected" [routerLinkActiveOptions]="{ exact: false }">
-    <a class="side-nav__item side-nav-header">
-      <svg class="icon">
-        <use xlink:href="../../assets/icons/icon-sprite.svg#icon-font-size"></use>
-      </svg>
-      Typography
+      *ngFor="let item of filteredShowcaseRoutes"
+      [routerLink]="item.path"
+      [class]="{ 'is-selected': item.active, 'is-hidden': item.hidden }"
```

Comment:
Consider if not rendering the link at all (using `*ngIf`) when filtering would be easier...

Reply:
Been down that path, but ran into a world of pain in combination with `routerLinkActive`. The router uses an index to keep state of the active link, so it yields an undesirable result if elements are removed.
This way is a bit more hand-held, but it's robust and explicit

Reply:
Update: removed the `hidden` concept entirely - as I am managing the active state myself, there is no longer an issue with `routerLinkActive`. I just plainly filter the list, so no conditional rendering at all